backend/addcorpus/serializers.py

```diff
@@ -121,13 +121,51 @@ def to_representation(self, value):
         return render_documentation_context(content)
 
 
+class CorpusConfigurationField(serializers.SlugRelatedField):
+    '''
+    Serialise a related corpus configuration as the corpus name.
+    '''
+
+    def __init__(self, **kwargs):
+        super().__init__('corpus__name', **kwargs)
+
+    def get_queryset(self):
+        return CorpusConfiguration.objects.all()
+
+    def to_representation(self, value: CorpusConfiguration) -> str:
+        return value.corpus.name
+
+
+class ChoiceOrderSerializer(serializers.ChoiceField):
+    '''
+    Variant on the ChoiceField that serialises to the index in the choices array.
+    '''
+    def __init__(self, choices, **kwargs):
+        super().__init__(choices=choices, **kwargs)
+        self.index = {value: i for i, (value, label) in enumerate(choices)}
+
+    def to_internal_value(self, data):
+        return self.choices[data]
+
+    def to_representation(self, value):
+        key = super().to_representation(value)
+        return self.index[key]
```

Comment:
Also slightly overengineered, I would suggest just using a calculated property on the model and serializing that, **BUT**: the reverse operation is nice, being able to send the index.